aten/src/ATen/Declarations.cwrap

```diff
@@ -2939,8 +2939,9 @@
         kwarg_only: True
       - THTensor* mat1
       - THTensor* mat2
-    - cname: spaddmm
-      sparse: yes
+    - sparse: True
```

Comment:
Does `sparse: True` do anything in ATen's Declarations.cwrap?

Reply:
Oh, oops, never mind. My fault for not reading through the whole PR>

Reply:
It does now, though `aten_dense_sparse` can replace its functionality if we wanted it to.

Reply:
We also have `aten_sparse`. How does `sparse` differ from `aten_sparse`?

Reply:
For a given option, `aten_sparse` says that the cname of the option can be used to call a dense function (TH/THC) and a sparse function (THS/THCS).

Right now `sparse=True` is only used for functions called on dense tensors that have sparse tensor arguments.  `sparse=True` makes the code generation go through a path that checks if `aten_dense_sparse` is a part of the option.

Reply:
Can we please start a doc that describes all possible values that can appear in `Declarations.yaml`?

Reply:
So does `sparse=True` just mean to search for `aten_dense_sparse` in options? If so, why not just search for `aten_dense_sparse` in the options?

Reply:
Yes, that's what it means. The [name of the function](https://github.com/zou3519/pytorch/blob/094c059e55ad9d60af701430d6b153c08a9081ac/aten/src/ATen/preprocess_declarations.py#L184) that does the searching for `aten_dense_sparse` implies that it was designed for more than just searching for `aten_dense_sparse`, but right now that's all it does. 

I'll change it to just search for `aten_dense_sparse` so we don't have to worry about the `sparse` option, then.

Reply:
Okay it turns out that `sparse=True` does something else: it tells the parser to append `THS` or `THCS` to the `cname`. Options declared with `aten_sparse` do this automatically but `aten_dense_sparse` doesn't.

At any rate, I'll change the searching in my code to search for `aten_dense_sparse` instead of `sparse=True` and keep `sparse=True` in the cwrap.